pkg/apis/resource/structured/namedresources/types.go

```diff
@@ -0,0 +1,100 @@
+/*
```

Comment:
For now, I think this model is sufficient to get us started.

As I mentioned in my comment on the following doc -- 7 of the 12 primary uses cases for NVIDIA GPUs are covered by this very simple model:
https://docs.google.com/document/d/1bDO11rEq_Yhpgpk5RN0VwnMLV1_2wNWvtOyM_QoIV_Y/edit?disco=AAABGKn3Z2s

I have ideas on how we can / should expand this model in future releases, but for now, keeping it simple will help us to focus on validating the overall framework for working with structured models without overcomplicating things.

My only comment would be about the name .... is `namesresources` to non-descriptive? I think in the KEP I had called it `NamedResourcesWithAttributes`, but this seems too long. Not sure what the right answer is here.

/cc @thockin @johnbelamaric @liggitt

Reply:
If we expect to add a "named resources without attributes", then the longer name would make sense. But we don't, so the shorter name IMHO is sufficient.